examples/graph/tool_requires/using_protobuf.rst

```diff
@@ -0,0 +1,230 @@
+
+There are libraries which could behave as a library and as a tool requirement, e.g., `protobuf <https://github.com/conan-io/conan-center-index/tree/master/recipes/protobuf>`__
+Those libraries normally contains headers/sources that compound the library itself, and, perhaps, an extra tools
+(compilers, shell scripts, etc.). Both parts are used in a different contexts, let's think of this scenario using
+*protobuf* for instance:
+
+* I want to create a library which includes a compiled protobuf message. The protobuf compiler (build context)
+  needs to be invoked at build time, and the library with the compiled *.pb.cc* file needs to be linked against
+  the protobuf library (host context).
+
+
+Given that, we should be able to use protobuf in build/host context in the same Conan recipe. Basically, your package
+should look like:
+
+.. code-block:: python
+
+    def requirements(self):
+        self.requires("protobuf/3.18.1")
+
+    def build_requirements(self):
+        self.tool_requires("protobuf/3.18.1")
+
+
+This is the way to proceed with any other library used in both contexts. Nonetheless, let's see a detailed example to see
+how the example looks like.
+
+Please, first, clone the sources to recreate this project, you can find them in the
+`examples2.0 repository <https://github.com/conan-io/examples2>`_ on GitHub:
+
+.. code-block:: shell
+
+    git clone https://github.com/conan-io/examples2.git
+    cd examples2/examples/graph/tool_requires/using_protobuf/myserver
+
+
+The structure of the project is the following:
+
+.. code-block:: text
+
+    ./
+    ├── conanfile.py
+    ├── CMakeLists.txt
+    ├── addressbook.proto
+    ├── host-armv8
+    ├── host-x86_64
+    └── src
+       └── myserver.cpp
+    └── include
+       └── myserver.h
+    └── test_package
+       ├── conanfile.py
+       ├── CMakeLists.txt
+       └── src
+           └── example.cpp
+
+
+The ``conanfile.py`` looks like:
+
+.. code-block:: python
+
+    from conan import ConanFile
+    from conan.tools.cmake import CMake, cmake_layout
+
+
+    class myserverRecipe(ConanFile):
+        name = "myserver"
+        version = "1.0"
+        package_type = "library"
+        settings = "os", "compiler", "build_type", "arch"
+        options = {"shared": [True, False], "fPIC": [True, False]}
+        default_options = {"shared": False, "fPIC": True}
+        generators = "CMakeDeps", "CMakeToolchain"
+        # Sources are located in the same place as this recipe, copy them to the recipe
+        exports_sources = "CMakeLists.txt", "src/*", "include/*", "addressbook.proto"
+
+        def config_options(self):
+            if self.settings.os == "Windows":
+                self.options.rm_safe("fPIC")
+
+        def configure(self):
+            if self.options.shared:
+                self.options.rm_safe("fPIC")
+
+        def requirements(self):
+            self.requires("protobuf/3.18.1", transitive_headers=True)
```

Comment:
Cant the library use protobuf internally?

Reply:
I need it for `test_package`. If I remove it:
```
[ 50%] Building CXX object CMakeFiles/example.dir/src/example.cpp.o
In file included from /Users/franchuti/develop/examples2/examples/graph/tool_requires/using_protobuf/myserver/test_package/src/example.cpp:4:
/Users/franchuti/.conan2/p/b/myser09621fde1ef62/p/include/addressbook.pb.h:10:10: fatal error: 'google/protobuf/port_def.inc' file not found
#include <google/protobuf/port_def.inc>
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
make[2]: *** [CMakeFiles/example.dir/src/example.cpp.o] Error 1
make[1]: *** [CMakeFiles/example.dir/all] Error 2
make: *** [all] Error 2
```